contribution-guide.md

```diff
@@ -0,0 +1,154 @@
+- [Submit your contribution](#submit-your-contribution)
+- [Accept the Contributor License Agreement](#accept-the-contributor-license-agreement)
+- [Propose changes to Ballerina](#propose-changes-to-ballerina)
+
+## Get started
+
+- Join the conversations at:
+
+    - [Ballerina-Dev Google Group](https://groups.google.com/forum/#!forum/ballerina-dev): to join the developer team mailing list to discuss Ballerina roadmap, features, issues that are in progress etc.
+    - [StackOverflow](https://stackoverflow.com/questions/tagged/ballerina): to get help with Ballerina. (Use the Ballerina tag for your questions there.) 
+    - [Slack](https://app.slack.com/client/T47EAEKB5/DF818FX9Q): for real-time communications with the team and community.
+
+- **IMPORTANT:** Send an email to security@ballerina.io to report sensitive security-related issues. For details, see the <a href="https://ballerina.io/security/">security policy</a>.
+
+- Submitting a bug is just as important as contributing to code. Create a new issue in the relevant GitHub repo using the below links, fill out all sections in the issue template, and submit the bug/issue report. 
+
+    >**Tip:** If you are unsure whether you have found a bug, search existing issues in the corresponding repo on GitHub and raise it in the [Ballerina-Dev Google Group](#https://groups.google.com/forum/#!forum/ballerina-dev).
+   - compiler, runtime, standard library, or tooling: <a href="https://github.com/ballerina-platform/ballerina-lang/issues">ballerina-lang</a> repo.
+   - language specification: <a href="https://github.com/ballerina-platform/ballerina-spec/issues">ballerina-spec</a> repo.
+   - website: <a href="https://github.com/ballerina-platform/ballerina-dev-website/issues">ballerina-dev-website</a> repo. 
+
+-  Start with GitHub issues that can be fixed easily:
+    - Browse issues labeled "good first issue" in the <a href="https://github.com/ballerina-platform/ballerina-lang/issues">ballerina-lang</a> repo.
+    - Use comments on the issue itself to indicate that you will be working on it and get guidance and help.
+
+## Set up the prerequisites
+1. Download [Ballerina](https://ballerina.io) and go through the [learning resources](https://ballerina.io/learn).
+2. Read the [Ballerina Code of Conduct](https://ballerina.io/code-of-conduct).
+3. Download the Java SE Development Kit (JDK) version 8 from one of the following locations:
+    - [Oracle](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
+    - [OpenJDK](http://openjdk.java.net/install/index.html) and install it.
+>**Note:** Set the JAVA_HOME environment variable to the path name of the directory into which you installed JDK.
+4. Download [Node.js](https://nodejs.org/en/download/) (version 8.9.x or the latest LTS release) and install it.
+5. Download [npm](https://www.npmjs.com/get-npm) (version 5.6.0 or later) and install it.
+
+## Set up the developer environment
```

Comment:
Shouldn't we swap this section with "Obtain the source code " section? The reason is below sections are asking to open the balleirna-lang-repo in IDEs , but the details on getting the code is not explained yet.

Reply:
Changed the order of the sections as follows.

- [Get started](#get-started)
- [Set up the Prerequisites](#set-up-the-prerequisites)
- [Obtain the source code](#obtain-the-source-code)
- [Build the project](#build-the-project)
- [Set up the developer environment](#set-up-the-developer-environment)
    - [Set up IntelliJ IDEA](#set-up-intellij-idea)
    - [Set up Eclipse](#set-up-eclipse)
- [Contribute to Ballerina grammar](#contribute-to-ballerina-grammar)
- [Submit your contribution](#submit-your-contribution)
- [Accept the Contributor License Agreement](#accept-the-contributor-license-agreement)
- [Propose changes to Ballerina](#propose-changes-to-ballerina)